### PR TITLE
Iss53 (support for censoring)

### PR DESCRIPTION
**Pull request by smeisler**
Reference to #53 

Adds support for basic scrubbing (simply removing volumes above a framewise displacement threshold) and optimized scrubbing (see [Power et al., 2014](https://www.sciencedirect.com/science/article/pii/S1053811913009117?casa_token=B2bFBxDZissAAAAA:ar8VT9FQpTbMeKBmrO9Kmy7oHFVUWNh8Q9rw0K8nTh0xGQhM-cM4jU7RyhcDW0G8IR6rqI4BcXPv)).

### CODE REVIEW

**Review by pbellec — COMMENTED** [review]
The code looks great, thanks for the PR. 
I've made a few minor comments. 

The main component missing is a test. 
I don't think it's necessary to test directly the impact of censoring regressors, but at least check that regressors get correctly generated. A little bit in the spirit of this test: https://github.com/SIMEXP/load_confounds/blob/master/load_confounds/tests/test_parser.py#L210

File: load_confounds/parser.py [hunk 1]
```diff
@@ -139,6 +139,35 @@ def _pca_motion(confounds_motion, n_components):
     motion_pca.columns = ["motion_pca_" + str(col + 1) for col in motion_pca.columns]
     return motion_pca
 
+def _load_censoring(confounds_raw, censoring, fd_thresh):
+    """Perform basic censoring - Remove volumes if framewise displacement exceeds threshold"""    
+    n_scans = len(confounds_raw)
+    # Get indices of fd outliers
+    fd_outliers = np.where(confounds_raw['framewise_displacement'] > fd_thresh)[0]
```
[the remark this answers is not in the document]
Comment (smeisler) [reply load_confounds/parser.py:146]: You are right about DVARS (outlier is > 3%). Should I make this a variable threshold like FD or standardize it? Adding DVARS will have no impact on the code for optimized scrubbing.

File: load_confounds/parser.py [hunk 2]
```diff
@@ -139,6 +139,35 @@ def _pca_motion(confounds_motion, n_components):
...
+    # Do optimized scrubbing if desired
+    if censoring == 'optimized':
+        fd_outliers = _optimize_censoring(fd_outliers, n_scans)
+    # Make one-hot encoded motion outlier regressors
+    motion_outlier_regressors = pd.DataFrame(np.transpose(np.eye(n_scans)[fd_outliers]).astype(int))
```
[the remark this answers is not in the document]
Comment (smeisler) [reply load_confounds/parser.py:151]: I don't think OneHotEncoder does what we would want it to do in this context.

**Comment by smeisler** [comment]
Added dvars and documentation

File: load_confounds/parser.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (pbellec) [reply load_confounds/parser.py:147]: yes absolutely

**Comment by pbellec** [comment]
There is a minor conflict with the ICA-AROMA PR that got merged, but it's very simple to resolve.

**Comment by pbellec** [comment]
so if I recap, this PR is still missing a parameter for `thresh_dvars`. This would have different defaults for the optimized and non-optimized variants.

Also needs a test of generation of the censoring regressors. 

Let me know if you are OK with addressing this. I can also merge the PR and address these remaining issues myself.

**Comment by smeisler** [comment]
I suppose some strategies could also be added to account for the most common applications of basic and optimized scrubbing. But yes, what you said is largely what remains for the parser. I can take a stab at addressing it, but I haven't thought much about making the tests; I took a look at the example you pointed to and it wasn't immediately clear how to make one for my PR.

**Comment by smeisler** [comment]
Added the std_dvars_threshold.

**Comment by pbellec** [comment]
LGTM. This is great! Huge thanks for the PR. 

For the test I am going to include a test data later that features despiking regressors. 
I'll compare the implementation we have against the regressors of fMRIprep. 
This will be done in a separate PR I will draft, and will ask for your feedback to check the tests make sense.

@all-contributors add @smeisler for code, doc and idea

**Pull request merged**